everest-testing/src/everest/testing/core_utils/everest_core.py

```diff
@@ -159,11 +162,23 @@ def start(self, standalone_module: Optional[str] = None, modules_to_test: List[T
 
         expected_status = 'ALL_MODULES_STARTED' if standalone_module == None else 'WAITING_FOR_STANDALONE_MODULES'
 
-        status = status_listener.wait_for_status(STARTUP_TIMEOUT, [expected_status])
+        status = self.status_listener.wait_for_status(STARTUP_TIMEOUT, [expected_status])
         if status == None or len(status) == 0:
             raise TimeoutError("Timeout while waiting for EVerest to start")
 
         logging.info("EVerest has started")
+        if expected_status == 'ALL_MODULES_STARTED':
+            self.all_modules_started_event.set()
+        else:
+            def read_fifo():
+                if self.status_listener.wait_for_status(10, ["ALL_MODULES_STARTED"]):
+                    self.all_modules_started_event.set()
+                    logging.info("set all modules started event...")
+            trd = threading.Thread(target=read_fifo)
+            trd.start()
+
+    def wait_for_all_modules_started(self, timeout=20) -> bool:
```

Comment:
Who is using this function?

Reply:
I've been using it in some tests, you kind of need it for any test with the PyProbeModule if you only want to interact with it when the global ready has been triggered

Reply:
I've removed it and will be using the status_listener in my tests directly